packages/job-worker/src/playout/playout.ts

```diff
@@ -981,36 +939,92 @@ export async function onPartPlaybackStopped(context: JobContext, data: OnPartPla
 			if (!playlist) throw new Error(`RundownPlaylist "${data.playlistId}" not found!`)
 		},
 		(cache) => {
-			const playlist = cache.Playlist.doc
-			if (!playlist) throw new Error(`RundownPlaylist "${data.playlistId}" not found!`)
+			_onPartPlaybackStopped(context, cache, data)
+		}
+	)
+}
+function _onPartPlaybackStopped(context: JobContext, cache: CacheForPlayout, data: OnPartPlaybackStoppedProps) {
+	const playlist = cache.Playlist.doc
+	if (!playlist) throw new Error(`RundownPlaylist "${data.playlistId}" not found!`)
 
-			// This method is called when a part stops playing (like when an auto-next event occurs, or a manual next)
-			const rundowns = cache.Rundowns.findFetch({ playlistId: playlist._id })
+	// This method is called when a part stops playing (like when an auto-next event occurs, or a manual next)
+	const rundowns = cache.Rundowns.findFetch({ playlistId: playlist._id })
 
-			const partInstance = cache.PartInstances.findOne({
-				_id: data.partInstanceId,
-				rundownId: { $in: rundowns.map((r) => r._id) },
-			})
+	const partInstance = cache.PartInstances.findOne({
+		_id: data.partInstanceId,
+		rundownId: { $in: rundowns.map((r) => r._id) },
+	})
 
-			if (partInstance) {
-				// make sure we don't run multiple times, even if TSR calls us multiple times
+	if (partInstance) {
```

Comment:
Another method that could benefit from inverting the if-statement

Reply:
ref: https://github.com/nrkno/sofie-core/pull/737#discussion_r912742322